api/ocm/extensions/repositories/genericocireg/accessmethod_localblob.go

```diff
@@ -111,3 +124,107 @@ func (m *localBlobAccessMethod) Get() ([]byte, error) {
 func (m *localBlobAccessMethod) MimeType() string {
 	return m.spec.MediaType
 }
+
+////////////////////////////////////////////////////////////////////////////////
+
+type composedBlock struct {
+	m    *localBlobAccessMethod
+	refs []string
+}
+
+var _ blobaccess.DataAccess = (*composedBlock)(nil)
+
+func (c *composedBlock) Get() ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	for _, ref := range c.refs {
+		var finalize finalizer.Finalizer
+
+		_, data, err := c.m.namespace.GetBlobData(digest.Digest(ref))
+		if err != nil {
+			return nil, err
+		}
+		finalize.Close(data)
+		r, err := data.Reader()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
The finalizer logic here is misleading. This actually adds a finalizer that later will be called and not actually closing the reader. This logic should be apparent from the API. Call it `AddCloser` or something like that.

On top of that Close has a side-effect AND returns the finalizer which isn't being used here. Please try to stick with one paradigm in a method.

Reply:
The return value is intended to be chainable.

The Finalizer is not touched, here and therefore the interface should not be changed in the PR.
Anyway: `finalize.Close(x)` directly say what is done. The finalizer will close the given object.
Maybe, we could name the method `CloseOnFinalize`. Or, to be consistent with other other such methods `WithCloser`.

Reply:
Hum, either is okay. But yeah, don't have to do that here I guess. :)